src/ucp/core/ucp_rkey.c

```diff
@@ -40,6 +42,8 @@ void ucp_rkey_packed_copy(ucp_context_h context, ucp_md_map_t md_map,
     *(ucp_md_map_t*)p = md_map;
     p += sizeof(ucp_md_map_t);
 
+    *((uint8_t *)p++) = mem_type;
```

Comment:
pls use `uct_memory_type_t`  instead of `uint8_t` for casting and `sizeof` where relevant

Reply:
@evgeny-leksikov uct_memory_type_t is defined as enum.  wanted to save on rkey packed size with typecasting to uint8_t.

Reply:
Ok, then pls check if we have static assert for UCT_MD_MEM_TYPE_LAST <= 255 or similar

Reply:
added assert while packing rkey